src/transformers/pipelines/__init__.py

```diff
@@ -317,7 +326,7 @@
 # any tokenizer/feature_extractor might be use for a given model so we cannot
 # use the statically defined TOKENIZER_MAPPING and FEATURE_EXTRACTOR_MAPPING to
 # see if the model defines such objects or not.
-MULTI_MODEL_CONFIGS = {"VisionTextDualEncoderConfig", "SpeechEncoderDecoderConfig"}
+MULTI_MODEL_CONFIGS = {"SpeechEncoderDecoderConfig", "VisionEncoderDecoderConfig", "VisionTextDualEncoderConfig"}
```

Comment:
@Narsil, 

Is this correct way of adding a model which can have multiple tokenizers?

For example, [this model](https://huggingface.co/nlpconnect/vit-gpt2-image-captioning) uses `GPT2Tokenizer` as a tokenizer whereas [this one](https://huggingface.co/microsoft/trocr-base-printed) uses `RobertaTokenizer` but both are vision encoder decoders.

Reply:
`AutoTokenizer` should be able to handle it by looking at the config.

There is afaik only 1 type of model which actually uses really two different tokenizers for the same model, which are the Marian variants. It's dealt by swapping the underlying tokenizer when necessary (not great, but it does the trick).

`MULTI_MODEL_CONFIGS` are extremely general since the idea is just to use in coordination two different kind of models, that's why the pipeline needs to know if we're looking at such models, because for such models, we need to try and load `tokenizer` (which should work out of the box) without really looking too closely at the arch. (We could do that everywhere, but this auto loading was written this way because some models were misconfigured and this was the only way to properly load the pipeline despite misconfigurations).